internal/approver/evaluator/identity.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package evaluator
+
+import (
+	"crypto/x509"
+	"fmt"
+	"strings"
+)
+
+// validateIdentity validates that the SPIFFE ID contained in the X.509
+// certificate request matches that in the username.
+func (i *internal) validateIdentity(csr *x509.CertificateRequest, username string) error {
+	split := strings.Split(username, ":")
+	if len(split) != 4 || split[0] != "system" || split[1] != "serviceaccount" {
```

Comment:
Suggestion: as I understand this will be the SA of the pod that requested the volume, because the CSI Driver impersonated it when creating the CR. Would it make sense to add a brief comment about that here?

Reply:
Sure thing.